### DISCUSSION TITLE
What's the point of `literalWriter`?

### DISCUSSION THREAD

**Discussion by MilkeeyCat** [a-h/templ#942] [General]
I was checking code generation for dev mode and I don't really understand what's the point of using [literalWriter](https://github.com/a-h/templ/blob/a4638e3f5f63acf8550f96d108a663990ffe5185/generator/rangewriter.go#L28-L32) and `_templ.txt` files. How is it faster than just putting string literals in right in generated go files, as in production mode?

**Comment by a-h** [comment]
In watch mode, the code reads the file from disk, so you don't need to recompile and restart your Go program if you only make changes to text or HTML, see https://templ.guide/commands-and-tools/live-reload/ for more information - the sequence diagram should make it clearer how that works.

There is a plan to make it so that the code that's generated in prod mode is identical in dev mode, and to use an environment variable to switch to dev mode.

**Comment by MilkeeyCat** [comment]
Ah, I see, I didn't try to think about it like that, thanks!
